game/src/engine/audio/AudioService.cpp

```diff
@@ -22,75 +25,359 @@ const ALCchar* kDefaultDevice = nullptr;
 ALCint kContextAttributes = 0;
 
 // audio directories
-// note: since diagetic sound (3d sound) can only be in mono
-// we'll want all sfx to be mono too (music is fine in stereo)
 const std::string kSfxDirectory = "resources/audio/sfx/";
 const std::string kMusicDirectory = "resources/audio/music/";
 
+// the current music file to stream from
+AudioFile kMusicAudioFile;
+
+// just keeps track of how much of the file was played
+ALsizei kMusicPlayhead = 0;
+
+// number of buffers to use when streaming music files
+const std::size_t kStreamBufferAmount = 4;
+const ALsizei kStreamBufferSize = 65536;  // 32kb per buffer
```

Comment:
These could be `static constexpr` to make them compile time constants only for this file instead of just general constant variables

Reply:
ooh, got it! wasn't actually aware of `constexpr` until just right now lol

Reply:
Haha yep, I also didn't know it existed for the longest time :P some modern C++ stuff has a tendency to be a bit more hidden sadly

Fun fact, there's also an `if constexr (...) {...}` which can sometimes replace an `#ifdef`.